### PR TITLE
Weekly Patch Release v1.5.5

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Contains all bugfixes to be released next week. As bugfixes get merged to master, they will get cherry-picked into this branch by the individual author (core) or on-caller.

For staff/core: The new [release guide is here](https://www.notion.so/gridai/Release-Guide-ab132bf94f304056afc8ff992304c18f).

```
gh pr list -s merged -S 'merged:2021-12-01T18:30:00.000Z..2021-12-07T18:30:00.000Z' --json mergedAt,milestone,url,mergeCommit,title --jq 'sort_by(.mergedAt) | reverse | .[] | select((.milestone.title == 
"1.5.x") or (.milestone.title == null)) | [.mergedAt, .url, .mergeCommit.oid, .title] | join(" ")' --limit 100

``` 

```

2021-12-06T21:48:32Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10961 629ca09e0971d2bd31be13b443bcebfa268a2b52 fix TypeError cause failure in singal_connector teardown
2021-12-06T13:41:19Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10935 3e1f8aa312d15f35aa274d65859d261eff7698c9 Fix spawn plugins not deleting temp checkpoint
2021-12-06T06:31:39Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10836 6599ced17d939900c9cffdbf5a856af74d54f2d5 Don't import torch_xla.debug for torch-xla<1.8
2021-12-05T06:55:47Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10932 3d59a2faff72aa34254b156607458f2b9acf8523 Trigger TPU CI only on branches with a pull request
2021-12-03T17:54:05Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10917 7792b77932791b40beb72946094eb1d50f58981c Resolve: 'DummyExperiment' object does not support item assignment
2021-12-03T16:37:40Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10921 d92ab96f17db11386da41b9577bd8104742b79df Simplify some ddp-spawn tests
2021-12-03T14:34:47Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10810 8ba3b383c01a7aaf84f456fa6b2ae2b13f406080 Fix filteration logic for eval results
2021-12-03T13:51:35Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10915 553d429ecb5e832c83356672df54056c92704313 Fix `prepare_data` implementation in `BoringDataModule`
2021-12-02T11:46:57Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10889 ff20af41fd363f9520f5e0083d7a49051353cd1c Fix mathjax in RTD build
2021-12-02T10:36:11Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10870 c55bc433ce9b24ac774b9fddf702fbafa3e0677f Fix retrieval of batch indices when dataloader num_workers > 0
2021-12-02T09:05:32Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10739 541b983b9073b54e4b736c880fcc8039a43818b3 Fix deadlinks in docs
2021-12-02T08:52:50Z https://github.com/PyTorchLightning/pytorch-lightning/pull/10837 5b9995da046995919b49f984e672d7cc0658b93d Fix schedule reset logic in pytorch profiler
```

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Part of #1

cc @borda

### CODE REVIEW

**Comment by awaelchli** [comment]
Awesome, thanks @rohitgr7 for opening it!

**Comment by awaelchli** [comment]
I will mark it ready for review already, to enable the full CI to run here.

**Review by awaelchli — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM!

**Review by kaushikb11 — APPROVED** [review]

**Review by Borda — APPROVED** [review]
